week7/answers.sql

```diff
@@ -1,3 +1,309 @@
+		i.damage
+	FROM teams t
+		INNER JOIN team_members tm
+			ON t.team_id = tm.team_id
+		INNER JOIN inventory inv 
+			ON tm.character_id = inv.character_id
+		INNER JOIN items i
+			ON inv.item_id = i.item_id
+UNION 
+	SELECT
+		t.team_id,
+		t.name AS team_name,
+		i.name AS item_name,
+		i.armor,
+		i.damage
+	FROM teams t
+		INNER JOIN team_members tm
+			ON t.team_id = tm.team_id
+		INNER JOIN equipped eq
+			ON tm.character_id = eq.character_id
+		INNER JOIN items i
+			ON eq.item_id = i.item_id
+		ORDER BY team_id, item_name;
+
+
+DELIMITER ;;
+
+CREATE FUNCTION armor_total(char_id INT UNSIGNED)
+RETURNS INT UNSIGNED
+DETERMINISTIC
+BEGIN
+	DECLARE armor_points_equipped INT UNSIGNED;
+    DECLARE armor_points_cs INT UNSIGNED;
+    
+    SELECT SUM(i.armor) INTO armor_points_equipped
+		FROM characters c
+			INNER JOIN character_stats cs
+				ON c.character_id = cs.character_id
+			INNER JOIN equipped eq
+				ON c.character_id = eq.character_id
+			INNER JOIN items i
+				ON eq.item_id = i.item_id
+	WHERE c.character_id = char_id;
+    
+	SELECT cs.armor INTO armor_points_cs
+		FROM characters c
+			INNER JOIN character_stats cs
+				ON c.character_id = cs.character_id
+	WHERE c.character_id = char_id;
+        
+	return armor_points_equipped + armor_points_cs;
+END;;
+DELIMITER ;
+
+DELIMITER ;;
+CREATE PROCEDURE attack(IN char_attacked_id INT UNSIGNED, IN id_item_equipped INT UNSIGNED)
+BEGIN
+    	DECLARE char_armor INT SIGNED;
+    DECLARE char_health INT;
+	DECLARE attack_damage INT SIGNED;
+    DECLARE difference INT;
+    DECLARE new_char_health INT;
+    
+	-- character armor 
+    SELECT armor_total(char_attacked_id) INTO char_armor;
+    
+    -- character_health
+    SELECT health INTO char_health FROM character_stats WHERE character_id = char_attacked_id;
+    
+    
+    -- item damage and attack damage - character armor
+    SELECT i.damage INTO attack_damage
+        FROM items i
+	    INNER JOIN equipped eq
+	        ON eq.item_id = i.item_id
+    WHERE eq.equipped_id = id_item_equipped;
+
+    SET difference = attack_damage - char_armor;
+    SET new_char_health = char_health - difference;
+    
+    -- check what happens with item damage
+    IF difference > 0 THEN
+	    UPDATE character_stats SET health=new_char_health WHERE character_id = char_attacked_id;
+    ELSEIF difference >= char_health THEN
+	    DELETE FROM characters WHERE character_id = char_attacked_id;
+    END IF;
+    
+END;;
+DELIMITER ;
+
+DELIMITER ;;
+CREATE PROCEDURE equip(IN item_inventory_id INT UNSIGNED)
+BEGIN
+	DECLARE char_id INT UNSIGNED;
+    DECLARE inventory_item INT UNSIGNED;
+    DECLARE char_equip INT UNSIGNED;
+    DECLARE item_equip INT UNSIGNED;
+    
+    SELECT character_id INTO char_id FROM inventory WHERE inventory_id = item_inventory_id;
+    
+    SELECT item_id INTO inventory_item FROM inventory WHERE inventory_id = item_inventory_id;
+ 
+    SELECT character_id INTO char_equip FROM characters WHERE character_id = char_id;
```

Comment:
You already have the character ID. This statement is true: `char_id == char_equip`.